### PR TITLE
Release 1.2.38

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4169](https://jira.cms.gov/browse/MAT-4169)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: react/components/MadieConfirmDialog/index.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import PropTypes from "prop-types";
+import ErrorIcon from "@mui/icons-material/Error";
+import MadieDialog from "../MadieDialog";
+
+const MadieConfirmDialog = ({
```
Comment (RohitKandimalla) [inline react/components/MadieConfirmDialog/index.jsx:6]: Looks like this component is similar to MADiEDeleteDialog and MADiEDiscardDialog. Not sure why we need these many reusable components which are built on reusable dialog, but I see that we could reduce some over head with these dialogs, so I was wondering if we can generalize this MadieConfirmDialog not to be tied to a specific text or specific headers and specific button names, we could make them dynamic and use props to add fill the content ?

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: react/components/MadieConfirmDialog/index.jsx (hunk 1 above)
Comment (RohitKandimalla) [reply react/components/MadieConfirmDialog/index.jsx:6]: How about lets not modify MADiEDeleteDialog and MADiEDiscardDialog so no breaking changes, but lets just modify MadieConfirmDialog is be more generic?

Comment (sb-cecilialiu) [reply react/components/MadieConfirmDialog/index.jsx:6]: @RohitKandimalla Sure, that was my thought too. I think the button names, the main message and the warning message are already generic.

Comment (sb-cecilialiu) [reply react/components/MadieConfirmDialog/index.jsx:6]: @RohitKandimalla done

**Review by RohitKandimalla — APPROVED** [review]
